### PR TITLE
Initial version of Pmod SSR driver with basic on/off control via GPIO

### PR DESCRIPTION
**Pull request by bchassoul**
This PR introduces the initial version of the Pmod SSR (Solid State Relay) driver for GRiSP boards (`grisp-base` and `grisp2`). The driver enables control of a solid-state relay using a single GPIO pin by switching the relay on `pmod_ssr:on/0` and off `pmod_ssr:off/0`.

Here you can find the Pmod [reference manual](https://digilent.com/reference/pmod/pmodssr/reference-manual)

### CODE REVIEW

**Review by GwendalLaurent — CHANGES_REQUESTED** [review]
Nice PR 👍 

My comments are just small details

File: src/pmod_ssr.erl [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+% To turn on, use:
+% ```
+% 2> pmod_ssr:on().
+% ok
+% '''
+% To turn back off, use:
+% ```
+% 3> pmod_ssr:off().
+% ok
+% '''
```
[the remark this answers is not in the document]
Comment (bchassoul) [reply src/pmod_ssr.erl:58]: Good point, I made the following test with the `pmod_ssr` connected:
```
(oscillator@grisp_oscillator)1> pmod_ssr:on().
pmod_ssr:on().
ok
(oscillator@grisp_oscillator)2> pmod_ssr:on().
pmod_ssr:on().
ok
(oscillator@grisp_oscillator)3> pmod_ssr:on().
pmod_ssr:on().
ok
(oscillator@grisp_oscillator)4> pmod_ssr:off().
pmod_ssr:off().
ok
(oscillator@grisp_oscillator)5> pmod_ssr:off().
pmod_ssr:off().
ok
(oscillator@grisp_oscillator)6> ```

File: src/pmod_ssr.erl [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+
+%--- Callbacks -----------------------------------------------------------------
+
+% @private
+init(Slot) ->
+    % TODO: expand for more possible Slots
+    case {grisp_hw:platform(), Slot} of
+        {grisp_base, gpio1} -> ok;
+        {grisp2, gpio1} -> ok;
```
[the remark this answers is not in the document]
Comment (bchassoul) [reply src/pmod_ssr.erl:76]: ah very nice! yes that would allow to connect over `gpio2` in `grisp_base`

File: src/pmod_ssr.erl [hunk 3]
```diff
@@ -72,8 +72,8 @@ off() ->
 init(Slot) ->
     % TODO: expand for more possible Slots
     case {grisp_hw:platform(), Slot} of
-        {grisp_base, gpio1} -> ok;
-        {grisp2, gpio1} -> ok;
+        {_, gpio1} -> ok;
+        {grisp_base, gpio2} -> ok;
         {P, S} -> error({incompatible_slot, P, S})
     end,
     Pin1 = grisp_gpio:open(gpio1_1, #{mode => {output, 0}}),
```
Comment (GwendalLaurent) [inline src/pmod_ssr.erl:79]: If the user uses the Slot `{grisp_base, gpio2}` . You will still open `gpio1_1` for `Pin1` but if I understand the docs correctly, you should open `gpio2_1` instead

**Review by GwendalLaurent — CHANGES_REQUESTED** [review]

File: src/pmod_ssr.erl (hunk 1 above)
[the remark this answers is not in the document]
Comment (GwendalLaurent) [reply src/pmod_ssr.erl:58]: Did you observe anything special/unexpected on the hardware side ? (e.g. between the first and second `on/0` the current briefly stopped)

**Pull request merged**